File: 프로그래머스/unrated/181188. 요격 시스템/요격 시스템.py
def solution(targets):
    answer = 0
    targets.sort()
    end = targets[0][1]
    for s,e in targets[1:]:
        if s >= end: # 떨어진 경우
            answer += 1
            end = e
        elif e <= end: # 포함된 경우
            if e < end:
                end = e
    return answer+1

File: 프로그래머스/unrated/181188. 요격 시스템/요격 시스템.java
import java.util.*;
class Solution {
    public int solution(int[][] targets) {
        int answer = 0;
        int limit = 0;
        Arrays.sort(targets,Comparator.comparing((int[] a) -> a[0]).thenComparing((int[] a) -> a[1]));
        for (int[] t: targets){
            int left = t[0];
            int right = t[1];
            if (right <= limit){
                limit = right;
            }
            else if (left >= limit){
                answer ++;
                limit = right;
            }
        }
        
        //System.out.println(Arrays.deepToString(targets));
        
        return answer;
    }
}